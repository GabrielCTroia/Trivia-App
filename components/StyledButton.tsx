import React from 'react';
import {
  StyleSheet,
  TouchableOpacity,
  View,
  ViewStyle,
  TouchableOpacityProps,
  Text,
} from 'react-native';

interface IProps extends TouchableOpacityProps {
  title: string;
  style?: ViewStyle,
}

export class StyledButton extends React.Component<IProps> {
  render() {
    return (<TouchableOpacity onPress={this.props.onPress}>
      <View style={[styles.container, this.props.style]}>
        <Text style={styles.text}>{this.props.title}</Text>
      </View>
    </TouchableOpacity>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    padding: 10,
    alignContent: 'center',
  },

  text: {
    textTransform: 'uppercase',
    textAlign: 'center',
    fontSize: 20,
  }
})