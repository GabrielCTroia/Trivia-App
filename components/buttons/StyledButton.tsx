import React, { FunctionComponent } from 'react';
import {
  StyleSheet,
  TouchableOpacity,
  ViewStyle,
  TouchableOpacityProps,
  Text,
  TextStyle,
} from 'react-native';
import * as Colors from '../../styles/Colors';


export enum ButtonTypes {
  clear = 'clear',
  success = 'success',
  positive = 'positive',
  warning = 'warning',
  danger = 'danger',
}

type StyledButtonProps = TouchableOpacityProps & {
  title: string;
  buttonType?: ButtonTypes;
  style?: ViewStyle;
  textStyle?: TextStyle;
}

export const StyledButton: FunctionComponent<StyledButtonProps> = ({ buttonType = ButtonTypes.success, ...props }) => (
  <TouchableOpacity
    onPress={props.onPress}
    style={[styles.container, styles[buttonType], props.style]}
  >
    <Text
      style={[styles.text, styles[`text${buttonType}`], props.textStyle]}>
      {props.title}
    </Text>
  </TouchableOpacity>
);

const styles = StyleSheet.create({
  container: {
    padding: 10,
    alignContent: 'center',
    borderRadius: 16,
  },
  text: {
    textTransform: 'uppercase',
    textAlign: 'center',
    fontSize: 20,
    color: Colors.baseTextColorInverted,
  },
  ['text' + ButtonTypes.clear]: {
    color: Colors.baseTextColor,
  },

  [ButtonTypes.success]: {
    backgroundColor: Colors.successColor,
  },

  [ButtonTypes.positive]: {
    backgroundColor: Colors.positiveColor,
  },

  [ButtonTypes.clear]: {

  },

  [ButtonTypes.warning]: {
    backgroundColor: Colors.warningColor,
  },

  [ButtonTypes.danger]: {
    backgroundColor: Colors.dangerColor,
  },
})