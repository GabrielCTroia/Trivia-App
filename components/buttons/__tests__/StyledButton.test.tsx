import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { StyledButton } from '../StyledButton';

it('renders correctly', () => {
  const tree = renderer.create(<StyledButton title="Snapshot Button" />).toJSON();

  expect(tree).toMatchSnapshot();
});
