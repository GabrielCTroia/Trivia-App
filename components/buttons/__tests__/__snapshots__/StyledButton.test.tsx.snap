// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignContent": "center",
          "padding": 10,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "textAlign": "center",
          "textTransform": "uppercase",
        }
      }
    >
      Snapshot Button
    </Text>
  </View>
</View>
`;
